models/mongo/groups.go

```diff
@@ -239,15 +239,168 @@ func (repo *groupsRepository) AcceptInvite(ctx context.Context, filter *models.O
+	}
+
+	invites := *group.Invites
+	return &invites[0], nil
+}
+
+func (repo *groupsRepository) ListInvites(ctx context.Context, filter *models.ManyInvitesFilter, lo *models.ListOptions) ([]*models.ListInvitesResult, error) {
+	invites := make([]*models.ListInvitesResult, 0)
+
+	mongoDocumentMatch := bson.D{}
+
+	if filter != nil {
+		if *filter.SenderAccountID != "" {
+			mongoDocumentMatch = append(mongoDocumentMatch, bson.E{Key: "invites.senderAccountId", Value: *filter.SenderAccountID})
+		}
+		if *filter.RecipientAccountID != "" {
+			mongoDocumentMatch = append(mongoDocumentMatch, bson.E{Key: "invites.recipientAccountId", Value: *filter.RecipientAccountID})
+		}
+		if *filter.GroupID != "" {
+			mongoDocumentMatch = append(mongoDocumentMatch, bson.E{Key: "_id", Value: *filter.GroupID})
+		}
+	}
+
+	idToMongoCondition := func(id *string, varIdentifier string) interface{} {
+		if *id != "" {
+			return bson.D{{Key: "$eq", Value: []interface{}{varIdentifier, id}}}
+		}
+		return "true"
+	}
+
+	invitesArrayFilterCondition := bson.D{
+		{Key: "$and",
+			Value: bson.A{
+				idToMongoCondition(filter.SenderAccountID, "$$invite.senderAccountId"),
+				idToMongoCondition(filter.RecipientAccountID, "$$invite.recipientAccountId"),
+			},
+		},
+	}
+
+	// Match only groups that matches the filter (which match the invites that were asked for)
+	matchQuery := bson.D{{Key: "$match", Value: mongoDocumentMatch}}
+
+	// In those documents, filter the invites array to have only the invites that matches the filter
+	filterQuery := bson.D{{
+		Key: "$project", Value: bson.D{
+			{Key: "invites", Value: bson.D{
+				{Key: "$filter", Value: bson.D{
+					{Key: "input", Value: "$invites"},
+					{Key: "as", Value: "invite"},
+					{Key: "cond", Value: invitesArrayFilterCondition},
+				}},
+			},
+			}}}}
+
+	// Separate every invite in it's specific array element
+	unwindQuery := bson.D{{
+		Key:   "$unwind",
+		Value: "$invites",
+	}}
+
+	// Offset and Limit
+	if lo.Limit == 0 { // NOTE: Don't know if I should put this in the endpoint code
+		lo.Limit = 20
+	}
+
+	paginationSkip := bson.D{{
+		Key:   "$skip",
+		Value: lo.Offset,
+	}}
+	paginationLimit := bson.D{{
+		Key:   "$limit",
+		Value: lo.Limit,
+	}}
+
+	// Reorder every variables to have a concise a pertinent element
+	projectionQuery := bson.D{{
+		Key: "$project",
+		Value: bson.D{
+			{Key: "recipientAccountId", Value: "$invites.recipientAccountId"},
+			{Key: "senderAccountId", Value: "$invites.senderAccountId"},
+			{Key: "id", Value: "$invites.id"},
+			{Key: "groupId", Value: "$_id"},
+			{Key: "_id", Value: 0},
+		},
+	}}
+
+	cur, err := repo.coll.Aggregate(ctx, mongo.Pipeline{matchQuery, filterQuery, unwindQuery, paginationSkip, paginationLimit, projectionQuery})
+	if err != nil {
+		return nil, repo.mongoFindErrorToModelsError(mongoDocumentMatch, lo, err)
+	}
+
+	err = cur.All(ctx, &invites)
+	if err != nil {
+		return nil, repo.mongoFindErrorToModelsError(mongoDocumentMatch, lo, err)
+	}
```

Comment:
All the MongoDB operations (such as `Find`, `FindOneAndUpdate`, `Delete`, etc) are encapsulated to better handle logging, errors, etc.
Maybe make a small encapsulation of `coll.Aggregate` in `repository.go` so that others can use it in the future?

Reply:
Will do